proposals/NNNN-add-result.md

```diff
@@ -0,0 +1,293 @@
+# Add Result to the Standard Library
+
+* Proposal: [SE-NNNN](NNNN-add-result.md)
+* Authors: [Jon Shier](https://github.com/jshier)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+* Implementation: [apple/swift#19982](https://github.com/apple/swift/pull/19982)
+
+## Introduction
+
+Swift's current error-handling, using `throws`, `try`, and `catch`, offers automatic and synchronous handling of errors through explicit syntax and runtime behavior. However, it lacks the flexibility needed to cover all error propagation and handling in the language. `Result` is a type commonly used for manual propagation and handling of errors in other languages and within the Swift community. Therefore this proposal seeks to add such a type to the Swift standard library.
```

Comment:
`async throw` functions would also relieve the mentioned pain points, maybe reference the other proposal draft that that seeks to introduce Coroutines? https://gist.github.com/lattner/429b9070918248274f25b714dcfc7619

Reply:
While something like `async throw` may address the automatic handling of errors for async work, it wouldn't address anything for the other use cases, and would likely still need a manually propagating version (like a `Future` returning a `Result`).